core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -340,6 +341,26 @@ class CivilizationInfo {
 
         if (isCurrentPlayer() || otherCiv.isCurrentPlayer())
             UncivGame.Current.settings.addCompletedTutorialTask("Meet another civilization")
+        
+        // If exactly one of the civilizations is a city state, we gain gold for meeting them 
+        if (!(isCityState() || otherCiv.isCityState()) || isCityState() && otherCiv.isCityState()) return
+        
+        val cityState = if (isCityState()) this else otherCiv
+        val majorCiv = if (isCityState()) otherCiv else this
+        
```

Comment:
A lot of things are done both ways and checked for each way.
We should make a "one-sided meeting", like "private CivInfo.activateMeetLogic(otherCiv)" and meetCivilization(otherCiv){ otherCiv.activateMeetLogic(this); activateMeetLogic(otherCiv) }